src/common/translation.cpp

```diff
@@ -1452,31 +1438,39 @@ wxString wxTranslations::GetBestTranslation(const wxString& domain,
 wxString wxTranslations::GetBestTranslation(const wxString& domain,
                                             const wxString& msgIdLanguage)
 {
-    wxString lang = GetBestAvailableTranslation(domain);
+    // Determine the best language, including the msgId language, which is always
+    // available because it is present in the code:
+    wxString lang = DoGetBestAvailableTranslation(domain, msgIdLanguage);
+
     if ( lang.empty() )
     {
-        wxArrayString available;
-        available.push_back(msgIdLanguage);
-        available.push_back(msgIdLanguage.BeforeFirst('_'));
-        lang = GetPreferredUILanguage(available);
-        if ( lang.empty() )
-        {
-            wxLogTrace(TRACE_I18N,
-                       "no available language for domain '%s'", domain);
-        }
-        else
-        {
-            wxLogTrace(TRACE_I18N,
-                       "using message ID language '%s' for domain '%s'", lang);
-        }
+        wxLogTrace(TRACE_I18N,
+                   "no available language for domain '%s'", domain);
+    }
+    else if ( lang == msgIdLanguage )
```

Comment:
It's not very important because it only affects the debugging message, but shouldn't we check for `lang == msgIdLanguage.BeforeFirst('_')` too here?

Reply:
Thanks, fixed in 039360a0